receiver/cloudflarereceiver/client.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cloudflarereceiver collects cloudflare logs into OTLP format.
+package cloudflarereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/cloudflarereceiver"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/cloudflarereceiver/internal/models"
+)
+
+type client interface {
+	MakeRequest(ctx context.Context, startTime string, endTime string) ([]*models.Log, error)
+	BuildEndpoint(startTime string, endTime string) string
+}
+
+var _ client = (*cloudflareClient)(nil)
+
+var defaultBaseURL = "https://api.cloudflare.com/client/v4"
+
+type cloudflareClient struct {
+	api      *cloudflare.API
+	cfg      *Config
+	endpoint string
+}
+
+func newCloudflareClient(cfg *Config, baseURL string) (client, error) {
+	var api *cloudflare.API
+	var err error
+	switch {
+	case cfg.Auth.XAuthEmail != "" && cfg.Auth.XAuthKey != "":
+		api, err = cloudflare.New(cfg.Auth.XAuthKey, cfg.Auth.XAuthEmail)
+	case cfg.Auth.APIToken != "":
+		api, err = cloudflare.NewWithAPIToken(cfg.Auth.APIToken)
+	default:
+		return nil, errInvalidAuthenticationConfigured
```

Comment:
can this be done in cfg.Validate() ?

Reply:
actually, it is done in there - so you don't need to check that here anymore. cfg.Validate() is called before the component is created.

Reply:
Updated, removed default case